src/main/java/net/snowflake/ingest/streaming/internal/DataValidationUtil.java

```diff
@@ -309,15 +335,13 @@ static String validateAndParseArray(String columnName, Object input, long insert
    * @return JSON object representing the input
    */
   static String validateAndParseObject(String columnName, Object input, long insertRowIndex) {
-    JsonNode jsonNode =
-        validateAndParseSemiStructuredAsJsonTree(columnName, input, "OBJECT", insertRowIndex);
-    if (!jsonNode.isObject()) {
+    final String result =
+        validateAndParseSemiStructured(columnName, input, "OBJECT", insertRowIndex);
+    if (!result.startsWith(JsonToken.START_OBJECT.asString())) {
       throw valueFormatNotAllowedException(columnName, "OBJECT", "Not an object", insertRowIndex);
     }
```

Comment:
nit: why shouldn't validateAndParse do this validation instead of expecting the callsite to do it?

Reply:
We do the validation here to reject incorrect user input we have to protect against, otherwise we would let the customer ingest incorrect data.